File: src/DrifterApps.Holefeeder.Services.Dto/CategoryDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using DrifterApps.Holefeeder.Common.Enums;

namespace DrifterApps.Holefeeder.Services.Dto
{
    public class CategoryDto
    {
        public string Id { get; }

        [Required]
        public string Name { get; }

        [Required]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public CategoryType Type { get; }

        public string Color { get; }

        public decimal BudgetAmount { get; }

        public bool Favorite { get; }

        public bool System { get; }

        public CategoryDto(string id, string name, CategoryType type, string color, decimal budgetAmount, bool favorite, bool system)
        {
            Id = id;
            Name = name;
            Type = type;
            Color = color;
            BudgetAmount = budgetAmount;
            Favorite = favorite;
            System = system;
        }
    }
}

File: src/Holefeeder.Ui/lib/models/api/category_dto.dart
import 'package:decimal/decimal.dart';
import 'package:json_annotation/json_annotation.dart';

part 'category_dto.g.dart';

@JsonSerializable()
class CategoryDto {
  final String id;
  final String name;
  final String color;
  final double budgetAmount;
  final bool favorite;
  const CategoryDto({
    required this.id,
    required this.name,
    required this.color,
    required this.budgetAmount,
    required this.favorite,
  });
  factory CategoryDto.fromJson(Map<String, dynamic> json) =>
      _$CategoryDtoFromJson(json);
  Map<String, dynamic> toJson() => _$CategoryDtoToJson(this);
}
